extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/CatalogResource.java

```diff
@@ -0,0 +1,747 @@
+  public Response updateTableDefn(
+      @PathParam("dbSchema") String dbSchema,
+      @PathParam("name") String name,
+      TableSpec spec,
+      @QueryParam("version") long version,
+      @Context final HttpServletRequest req
+  )
+  {
+
+    TableDefnRegistry tableRegistry = catalog.tableRegistry();
+    return incrementalUpdate(
+        TableId.of(dbSchema, name),
+        spec,
+        req,
+        (existing) -> tableRegistry.resolve(existing).merge(spec).spec()
+    );
+  }
+
+  private Response incrementalUpdate(
+      TableId tableId,
+      TableSpec newSpec,
+      @Context final HttpServletRequest req,
+      Function<TableSpec, TableSpec> action
+  )
+  {
+    Response response = authorizeTable(tableId, newSpec, req);
+    if (response != null) {
+      return response;
+    }
+    try {
+      long newVersion = catalog.tables().updatePayload(tableId, action);
+      return Actions.okWithVersion(newVersion);
+    }
+    catch (NotFoundException e) {
+      return Response.status(Response.Status.NOT_FOUND).build();
+    }
+    catch (Exception e) {
+      return Actions.exception(e);
+    }
+  }
+
+  /**
+   * Move a single column to the start end of the column list, or before or after
+   * another column. Both columns must exist. Returns the version of the table
+   * after the update.
+   * <p>
+   * The operation is done atomically so no optimistic locking is required.
+   *
+   * @param dbSchema
+   * @param name
+   * @param command
+   * @param req
+   * @return
+   */
+  @POST
+  @Path("/tables/{dbSchema}/{name}/moveColumn")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response moveColumn(
+      @PathParam("dbSchema") final String dbSchema,
+      @PathParam("name") final String name,
+      final MoveColumn command,
+      @Context final HttpServletRequest req
+  )
+  {
+    if (command == null) {
+      return Actions.badRequest(Actions.INVALID, "A MoveColumn object is required");
+    }
+    if (Strings.isNullOrEmpty(command.column)) {
+      return Actions.badRequest(Actions.INVALID, "A column name is required");
+    }
+    if (command.where == null) {
+      return Actions.badRequest(Actions.INVALID, "A target location is required");
+    }
+    if ((command.where == Position.BEFORE || command.where == Position.AFTER) && Strings.isNullOrEmpty(command.anchor)) {
+      return Actions.badRequest(Actions.INVALID, "A anchor column is required for BEFORE or AFTER");
+    }
+    return incrementalUpdate(
+        TableId.of(dbSchema, name),
+        null,
+        req,
+        (spec) -> spec.withColumns(command.perform(spec.columns()))
+    );
+  }
+
+  /**
+   * Hide or unhide columns. If both appear, hide takes precedence. Returns the
+   * new table version.
+   */
+  @POST
+  @Path("/tables/{dbSchema}/{name}/hideColumns")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response hideColumns(
```

Comment:
Seems that we are combining both hide and unhide here, should we have separate apis for each?